<launch>
    <!-- Запуск камеры -->
    <include file="$(find pidrone_pkg)/launch/raspicam_node.launch" />
    
    <!-- Запуск web_video_server для трансляции изображений в веб-интерфейс -->
    <node name="web_video_server" pkg="web_video_server" type="web_video_server" output="screen">
        <param name="port" value="8080" />
        <param name="quality" value="70" />
    </node>
    
    <!-- Запуск ноды распознавания ArUco маркеров -->
    <node name="recognition_of_aruco_marker" pkg="pidrone_pkg" type="recognition_of_aruco_marker_node.py" output="screen">
        <!-- Путь к файлу с картой маркеров -->
        <param name="map_path" value="$(find pidrone_pkg)/scripts/aruco/sample_map.txt" />
        
        <!-- Топик с изображением с камеры -->
        <param name="image_topic" value="/raspicam_node/image_raw" />
        
        <!-- Префикс для TF фреймов маркеров -->
        <param name="frame_id_prefix" value="aruco_" />
    </node>
</launch> 